### PR TITLE
ADD:implement set up react and rails application

### PR DESCRIPTION
**Pull request by davidigbo**
## In this PR, I setup react and rails applications by following these steps:

- [ ] Create a new Rails app called 'hello-rails-react'.
- [ ] Use the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
- [ ] Verify that the jsbundling-rails gem is added to your Gemfile.
- [ ] Initialize your project with Git.
- [ ] Install needed node dependencies as described in the previous lesson's tutorial.
- [ ] Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- [ ] Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
- [ ] Make sure that your project has a Postgres database set up.
- [ ] Create a table for storing your messages and create 5 different greetings.
- [ ] Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- [ ] Create a static view that will be the root of your app.
- [ ] Create the App component with react-router and render it in your static view.
- [ ] Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- [ ] Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- [ ] Display the random greeting in your Greeting component.

### CODE REVIEW

**Review by julie-ify — COMMENTED** [review]
Hi @david30-maker,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* You have submitted a project without a valid readme file

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -1,23 +1,153 @@
...
+
+
+<!-- Features -->
+
+</ul>
+
+<h1>🚀 Live Demo</h1>
+<a href="">Live Demo</a>
+<h1>Video presentation</h1>
+<a href=d="">Click here</a>
```
Comment (joel-kalema) [inline README.md]: - [ ] You did a great job, however, your links are broken 😿, make sure you fixed them to make your README more professional 👍

File: README.md [hunk 2]
```diff
@@ -1,23 +1,153 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](LICENSE.md) licensed.
+
```
Comment (joel-kalema) [inline README.md]: - [ ] Also there 👍

**Review by joel-kalema — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @david30-maker 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

0bd223a2b156e0eaf61a713e84c3d3b7



## To highlight :champagne: 

- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me @joel-kalema  when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 3]
```diff
@@ -1,23 +1,150 @@
...
+  </ul>
+</details>
+
+
+<!-- Features -->
+
+</ul>
+
+<h1>🚀 Live Demo</h1>
+
```
Comment (julie-ify) [inline README.md]: - [ ] Since you have no live demo link, then kindly remove this section and the link on the header.

File: README.md [hunk 4]
```diff
@@ -1,23 +1,150 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](LICENSE.md) licensed.
```
Comment (julie-ify) [inline README.md]: - [ ] The license link is still not fixed. The file you created needs to be properly linked.

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @david30-maker,

## Highlights!
- jsbundling for project

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Olamarx — APPROVED** [review]
Hi @david30-maker, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good commit history ✔️
- Good PR description ✔️
- Follow GitHub flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**